src/components/table.ts

```diff
@@ -155,23 +155,38 @@ export module Component {
       var nIterations = 0;
       while (true) {
         var offeredHeights = _Util.Methods.addArrays(guaranteedHeights, rowProportionalSpace);
-        var offeredWidths  = _Util.Methods.addArrays(guaranteedWidths,  colProportionalSpace);
+        var offeredWidths  = _Util.Methods.addArrays(guaranteedWidths, colProportionalSpace);
         var guarantees = this._determineGuarantees(offeredWidths, offeredHeights);
         guaranteedWidths = guarantees.guaranteedWidths;
         guaranteedHeights = guarantees.guaranteedHeights;
         var wantsWidth  = guarantees.wantsWidthArr .some((x: boolean) => x);
         var wantsHeight = guarantees.wantsHeightArr.some((x: boolean) => x);
 
-        var lastFreeWidth  = freeWidth ;
+        var lastFreeWidth  = freeWidth;
         var lastFreeHeight = freeHeight;
         freeWidth  = availableWidthAfterPadding  - d3.sum(guarantees.guaranteedWidths );
         freeHeight = availableHeightAfterPadding - d3.sum(guarantees.guaranteedHeights);
+
         var xWeights: number[];
         if (wantsWidth) { // If something wants width, divide free space between components that want more width
           xWeights = guarantees.wantsWidthArr.map((x) => x ? 0.1 : 0);
           xWeights = _Util.Methods.addArrays(xWeights, colWeights);
-        } else { // Otherwise, divide free space according to the weights
-          xWeights = colWeights;
+        } else {
+          if (d3.sum(colWeights) / colWeights.length !== colWeights[0] ||
+              d3.sum(colWeights) === 0) {
+            // Use colWeights if weights are not equal OR weights are all set to zero (fixed-width)
+            xWeights = colWeights;
+          } else {
+            // If they are all equal, divide free space to make components as equal as possible
```

Comment:
Shouldn't we divide the free space so that the *resulting sizes* are proportional to the weights whenever possible, rather than just if the weights are equal?